docs/architecture/README.md

```diff
@@ -1,15 +1,51 @@
 # Architecture
 
+This section covers the core concepts of the `cephlet` project.
+
+## Startup flow
+
+Upon startup, the `cephlet` controller manager announces it's `VolumePool` in the cluster as configured by the following
+flags:
+
+```shell
+--volume-pool-annotations  
+      Annotations to apply to the volume pool upon startup. (default [])
+--volume-pool-labels       
+      Labels to apply to the volume pool upon startup. (default [])
+--volume-pool-name         
+      The name of the volume pool. (default "ceph")
+--volume-pool-replication  
+      The replication factor of the volume pool. (default 3)
+```
+
+The `VolumePool` indicates his supported `VolumeClasses` in the `.status.availableVolumeClasses` status field. This is
+populated by the `VolumePoolReconciler` of the `cephlet`.
+
+```mermaid
+graph TD
+    C[cephlet] -- announce on startup --> VP[VolumePool]
+    C -. filter .-> VC[VolumeClasses]
+    VC -. available .-> VP
+```
+
+Which `VolumeClasses` should be supported by the announced `VolumePool` can be configured via the following flag
+
+```shell
+--volume-class-selector 
+        Selector for volume classes to report as available. (default [])
+```
+
 ## Volume Provisioning
 
+The main task of the `cephlet` is to create Ceph block devices for every `Volume` in a given namespace.
+
 !!! note
-    As Ceph does not support the creation of access secrets for each individual block devices, we chose to use Kubernetes `Namespace`s
-    as a tenant separation. All `Volumes` within a `Namespace` belong essentially to one tenant.
+    As Ceph does not support the creation of access secrets for each individual block devices, we chose to use 
+    Kubernetes `Namespace`s as a tenant separation. All `Volumes` within a `Namespace` belong essentially to one tenant.
 
-The task of the `cephlet` is to create a Ceph block device for every `Volume` in a given namespace. Additionally,
-for every `Namespace` an own `CephClient` and a corresponding `StorageClass` is created. The `StorageClass` is being used
-to later create the `PersistentVolumeClaims` for a given `Namespace`. The access credentials which are being extracted
-from the `CephClient` are stored in a `Secret` which is then referenced in the status of the `Volume`.
+Additionally, for every `Namespace` an own `CephClient` and a corresponding `StorageClass` is created. The `StorageClass` 
+is being used to later create the `PersistentVolumeClaims` for a given `Namespace`. The access credentials which are being 
+extracted from the `CephClient` are stored in a `Secret` which is then referenced in the status of the `Volume`.
```

Comment:
to simplify the cleanup we create for each volume a secret and copy the content of the ceph client secret. 
additionally: the ceph client secret is in the "ceph default ns"

Reply:
You are right I simplified this part here a bit.